test/e2e/cypress/e2e/hosts_overview.cy.js

```diff
@@ -170,4 +170,41 @@ context('Hosts Overview', () => {
       });
     });
   });
+
+  describe('Deregistration', () => {
+    describe('Clean-up buttons should be visible only when needed', () => {
+      const heartbeatTargets = agents();
+      const failingHost = heartbeatTargets.shift();
```

Comment:
`shift()` mutates the `heartbeatTargets` array, is this safe for the tests?

Reply:
That's exactly the intention, to leave that host out of the array so that it doesn't get a heartbeat